--- templates/SenchaCommon/Boot.dist.js	2018-06-11 04:19:07.505783722 +0000
+++ templates/SenchaCommon/Boot.js	2018-06-11 04:19:52.177343192 +0000
@@ -1113,6 +1113,7 @@
                     }
                     
                     entry = Boot.getEntry(url, {
+                        cache: me.cache,
                         buster: me.buster,
                         charset: me.charset
                     }, item && item.canonicalPath);
@@ -1828,4 +1829,4 @@
 
     output.push(path);
     return output.join('/');
-};
\ No newline at end of file
+};
